### PR TITLE
fix(storage): fix ConcatSstableIterator seek.

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Fix https://github.com/risingwavelabs/risingwave/issues/5518. 

- Use seek_key in seek and seek_idx.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
close https://github.com/risingwavelabs/risingwave/issues/5518

### CODE REVIEW

**Review by Li0k — APPROVED** [review]
LGTM, @Little-Wallace PTAL

File: src/storage/src/hummock/compactor/iterator.rs [hunk 1]
```diff
@@ -250,8 +228,11 @@ impl ConcatSstableIterator {
             let add = (now.elapsed().as_secs_f64() * 1000.0).ceil();
             stats_ptr.fetch_add(add as u64, atomic::Ordering::Relaxed);
 
-            let mut sstable_iter =
-                SstableStreamIterator::new(block_stream, end_index - start_index, &self.stats);
+            let mut sstable_iter = SstableStreamIterator::new(
```
[the remark this answers is not in the document]
Comment (zwang28) [reply src/storage/src/hummock/compactor/iterator.rs:231]: We've explicitly stopped at the end of key-range in https://github.com/risingwavelabs/risingwave/blob/9147b9fc54ba800399ccb90faee0647b1a8b21f1/src/storage/src/hummock/compactor/mod.rs#L547

[the remark this answers is not in the document]
Comment (Li0k) [reply src/storage/src/hummock/compactor/iterator.rs:231]: same with zwang, I think this would work if the iterator only works for the compactor logic

**Comment by ALeitert** [comment]
LGTM

**Review by ALeitert — APPROVED** [review]

**Review by Li0k — APPROVED** [review]
LGTM! thanks

**Pull request merged**